cirq-core/cirq/protocols/json_serialization.py

```diff
@@ -505,6 +513,88 @@ def get_serializable_by_keys(obj: Any) -> List[SerializableByKey]:
     return []
 
 
+def json_cirq_type(type_obj: Type) -> str:
+    """Returns a string type for JSON serialization of `type_obj`.
+
+    Types can provide custom string types with `_json_cirq_type_`; otherwise, a
+    Cirq type will use its type name as its string type. Non-Cirq types do not
+    have a json_cirq_type value.
+
+    This method is not part of the base serialization path. Together with
+    `cirq_type_from_json`, it can be used to provide type-object serialization
+    for classes that need it.
+
+    Args:
+        type_obj: a Type to be converted to its string identifier.
+
+    Returns:
+        A string which uniquely identifies `type_obj`.
+
+    Raises:
+        ValueError: if `type_obj` is not a Cirq type and does not explicitly
+            define its string representation with _json_cirq_type_.
+    """
+    if hasattr(type_obj, '_json_cirq_type_'):
+        return type_obj._json_cirq_type_()
+    if type_obj.__module__.startswith('cirq'):
+        return type_obj.__name__
+    raise ValueError(f'{type_obj} is not a Cirq type, and does not define _json_cirq_type_.')
```

Comment:
If there are e.g. classes in recirq that define `_json_dict_` but don't define `_json_cirq_type_`, would this value error get raised? or only if I use some of the (optional?) new functionality?

Reply:
Only if you use the new behavior, e.g. by attempting to serialize recirq types into an object that uses type serialization (in Cirq, this is only planned for `Gateset` and `GateFamily` so far)

Reply:
thanks